### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by mscherer**
Wanting to use that role, I found it was a bit derelict and that our standard and best practice did evolve a bit in 1 year and half. I bundled the whole set of change as 1 PR so it get faster to merge.

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -1,23 +1,12 @@
...
-
-- set_fact: rdbms_service=mysqld
-  when: rdbms == 'mysql'
-
-- set_fact: log_file=/var/log/mysql.log
-  when: rdbms == 'mysql'
-
-- set_fact: log_file=/var/log/mariadb/mariadb.log
-  when: rdbms == 'mariadb'
+- include_vars: "{{ rdbms }}.yml"
```
Comment (duck-rh) [inline tasks/main.yml]: What if I set rdbms to an existing unrelated file, like "main" for example?
I would suggest prefixing the files.

**Review by duck-rh — CHANGES_REQUESTED** [review]
Except for one small change I think we should do (see review), the rest is fine.

File: tasks/main.yml (hunk 1 above)
Comment (mscherer) [reply tasks/main.yml]: If the user decide to do something undocumented on purpose, I am not sure we can do much. What if the user then put a file "prefix_mysql.yml" in the search path and break the role ?

We can check that rdbms is set to the 2 authorized values, but that's a corner case, and blocking a review with both a important bug fix ( https://github.com/OSAS/ansible-role-mariadb/pull/2/commits/8b180ea1e5327f718192686105721dbda93dcc0d ) and a potential security issue ( https://github.com/OSAS/ansible-role-mariadb/pull/2/commits/be671c6ffb0e586aa33f48c45ef7f96ffed0dc8f ) on that make me wonder on the whole process. 

I will bring the topic during our meeting next week, because to me, this is bringing more issues than it prevent, and the process must be changed.

Comment (duck-rh) [reply tasks/main.yml]: well, if you want to merge an important fix, then you do not meld it in a bunch of low priority stuff.

Also we discussed this situation where is it urgent and IIRC we agreed on the role maintainer deciding for merging quickly without review or not, and then fixing better or discussing it later.

Comment (duck-rh) [reply tasks/main.yml]: as for this point, you can disagree and discard it as the maintainer. I just though it is different than the OS type because it is a "user" input and better have a "you entered a bad value" hint than doing something strange to debug.

**Review by duck-rh — APPROVED** [review]

File: tasks/main.yml (hunk 1 above)
Comment (mscherer) [reply tasks/main.yml]: I did 5 commits because I think that having 5 PR is not efficient. Either they all got merged right away (best case) and we have to make 5 review instead of 1 (so merging in order, hitting 10 times a button instead of 2, and if possible, do that without being interrupted). Or there is change to be done, and if that's on the first PR, we have to rebase, and if that's on the last, we are in the same amount of work as the case of merging all. 

In both case, this represent work and cognitive load. 

And the fix was urgent enough that I had to make my own branch, so I had a solution, but if I had to fork the software, and plan to do more and more because that provides me with faster feedback loops and less impediment (so faster velocity), I think we might need to rediscuss the workflow.

**Comment by mscherer** [comment]
So, since this one wasn't merged and since I merged your changes (instead of delaying), now I have to rediff all of it. I think that me doing the right thing (ie, trying to be fast to not block you), shouldn't result into me having to do much more work by rediffing and all. That's really piss me off.

**Comment by duck-rh** [comment]
making the CI work is worth it I think, but thanks anyway for the prioritizing my PRs.

**Comment by mscherer** [comment]
I think, it just that now, I have to rediff all, so I will do that when I have some downtime, cause that's likely too complex to be a few git commands. The patch is quite sizeable (so I can't just move things around to make it mergeable), and at the same time not important.

**Comment by duck-rh** [comment]
Sorry for the trouble. Would you like me to recreate PRs based on your ideas? (as you seem to be busy and annoyed)

In the same area, what would you think to put rdbms conditionals into OS-specific vars includes?

**Comment by mscherer** [comment]
Oh no, that's not urgent. I am just annoyed at myself for merging stuff that conflicted with my own, but that was expected for something that touch to everything. I will do that when I have enough free neurons to focus on it, but it work fine so there 0 urgency.

**Comment by mscherer** [comment]
So I finally did rediff everything, which was a pain since reformating had to be redone from 0. But now, it do apply fine.

**Comment by mscherer** [comment]
And it fail due to the Debian vars (since that's tested on Ubuntu precise). What should we do, ignore the tests or now, or is there a debian packager volunteer to fix ?

**Comment by duck-rh** [comment]
It's all fine.

The build fails for lack of support for Debian systems. It's up to you to bypass the tests or add the support and request an additional review.

**Comment by mscherer** [comment]
Or rather, wrong support on Ubuntu precise

**Comment by duck-rh** [comment]
I think the service name is without 'd' on Ubuntu, as I can see it on a Debian right now.

**Comment by mscherer** [comment]
So the last patch, because it fix a bug that did prevent stuff from working on Debian/Mysql do reveal another bug, awesome...

**Comment by duck-rh** [comment]
All good :-)
